### PR TITLE
Use new version of publish / staging plugin which has merged the two …

### PR DESCRIPTION
**Pull request by anuraaga**
…projects.

It's the year of projects merging together!
https://github.com/gradle-nexus/publish-plugin

### CODE REVIEW

File: gradle/publish.gradle [hunk 1]
```diff
@@ -102,7 +89,6 @@ private String artifactPrefix(Project p, String archivesBaseName) {
 }
 
 rootProject.tasks.release.finalizedBy tasks.publishToSonatype
-rootProject.tasks.closeAndReleaseRepository.mustRunAfter tasks.publishToSonatype
```
Comment (anuraaga) [inline gradle/publish.gradle:105]: This is handled by the plugin now

**Comment by anuraaga** [comment]
/cc @asarkar I randomly noticed you're looking into maven publishing plugins. Here's yet another one :P Our build doesn't function with any other plugin since we need the feature this one has of initializing the staging repository before uploading, instead of using the implicitly created repository and working on it. It's because with the count of our artifacts, Maven always flakes and creates a second staging repository during a single build if we don't initialize it explicitly first, causing other plugins to fail.

**Comment by asarkar** [comment]
@anuraaga 
> I randomly noticed you're looking into maven publishing plugins.

I have my fingers in many pies, can you refresh my memory?

**Review by iNikem — APPROVED** [review]

**Comment by anuraaga** [comment]
@asarkar https://github.com/DanySK/maven-central-gradle-plugin/issues/88

**Review by trask — APPROVED** [review]

**Review by mateuszrzeszutek — APPROVED** [review]

**Pull request merged**